presto-main/src/main/java/com/facebook/presto/operator/SimplePagesHashStrategy.java

```diff
@@ -39,14 +47,19 @@
     private final List<Integer> hashChannels;
     private final List<Block> precomputedHashChannel;
     private final Optional<Integer> sortChannel;
+    private final FunctionRegistry functionRegistry;
+    private final boolean groupByUsesEqualTo;
+    private final Map<Type, MethodHandle> methodHandles;
```

Comment:
nit: Name is too vague, maybe `isNotDistinctFromMethodHandles`?

Also, I would suggest just to use a `List` instead of `Map` here unless there are reasons we want to do so.